python_src/src/lamp_py/performance_manager/l0_gtfs_rt_events.py

```diff
@@ -111,11 +111,28 @@ def combine_events(
         "trip_stop_hash",
         "vehicle_id",
         "trip_id",
+        "vehicle_label",
+        "vehicle_consist",
     ]
 
     event_details = pandas.concat(
         [vp_events[details_columns], tu_events[details_columns]]
-    ).drop_duplicates(subset="trip_stop_hash")
+    )
+
+    # create sort columns to indicate which records have null values for vehicle_label or vehicle_consist
+    # we want to drop these null value records whenever possible
+    event_details["na_sort"] = (
+        event_details[["vehicle_label", "vehicle_consist"]].isna().sum(axis=1)
+    )
```

Comment:
When combining parameter columns from trip updates and vehicle positions, we want to prioritize vehicle positions records with `vehicle_label` and `vehicle_consist` values. This `na_sort` column will count the number of na's from those two columns and use them as a sort when duplicates are dropped from this record set.